src/embedding/kd.ts

```diff
@@ -332,6 +358,53 @@ export namespace EmbeddedKpiDashboard {
         return getEventType(obj) === GdcKdCommandType.SetSize;
     }
 
+    /**
+     * Data type of SetFilterContext command
+     */
+    export type SetFilterContextCommandData = IGdcKdMessageEnvelope<GdcKdCommandType.SetFilterContext, EmbeddedGdc.IFilterContextContent>;
+
+    /**
+     * Add or update the filter context
+     *
+     * Contract:
+     * - if filters is same with filters on the KD filter bar, then update the filters on the filter bar and apply the filters to dashboard
+     * - if filters is new one, then add it to the KD filter bar and apply to dashboard
+     * - in-case the AD can not apply the filters, a CommandFailed will be posted. The reason could be:
+     *   - exceed the limit number of filter items for selected filter values
+     */
+    export type SetFilterContextCommand = IGdcKdMessageEvent<GdcKdCommandType.SetFilterContext, EmbeddedGdc.IFilterContextContent>;
+
+    /**
+     * Type-guard checking whether an object is an instance of {@link SetFilterContextCommand}
+     *
+     * @param obj - object to test
+     */
+    export function isSetFilterContextCommandData(obj: any): obj is SetFilterContextCommandData {
+        return getEventType(obj) === GdcKdCommandType.SetFilterContext;
+    }
+
+    /**
+     * Data type of removeFilterContext command
+     */
+    export type RemoveFilterContextCommandData = IGdcKdMessageEnvelope<GdcKdCommandType.RemoveFilterContext, EmbeddedGdc.IFilterContextContent>;
+
+    /**
+     * Remove the filter context
+     * Contract:
+     * - if filters is in the filter bar, then update the filters on the filter bar and apply the filters to insight
+     * - if filters is not in the filter bar, then a CommandFailed will be posted.
```

Comment:
1. `filters are ...`
2. `then remove the filters on the filter bar and update insight`

Reply:
```
if filters existed on the KD filter bar, then the filters will be removed and the insight will be updated
if filters are in the KD filter bar, then a CommandFailed will be posted.
```